pydantic/dataclasses.py

```diff
@@ -190,6 +192,12 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         Returns:
             A Pydantic dataclass.
         """
+        if is_model_class(cls):
```

Comment:
I'd make the same suggestion re the test failures, let's see if this approach fixes things: https://github.com/pydantic/pydantic/pull/9398#discussion_r1600373980